lib/game.rb

```diff
@@ -0,0 +1,136 @@
+require_relative 'board'
+require_relative 'player'
+require_relative 'constants'
+
+class Game
+	attr_accessor :board, :turn, :player1, :player2
+	def initialize
+		@board = Board.new
+		@turn = 0
+		@player1 = nil
+		@player2 = nil
+	end
+
+	def add_player(symbol)
+		player_name = gets.chomp
+		return Player.new(player_name, symbol)
+	end
+
+	def move(player)
+		symbol = player.symbol
+		puts "#{player.name} move (print a number from 0 to 8) \n"
+		move_index = gets.chomp
+		if board.matrix[move_index.to_i] == nil && move_index.to_i >= 0 && move_index.to_i < 9 && POSSIBLE_MOVES.any? {|element| element == move_index} 
+			board.change_matrix(player.symbol, move_index.to_i)
+		else
+			puts 'Please make a valid move'
+			move(player)
+		end
+	end	
+
+	def board_full?
+		!board.matrix.any? {|val| val == nil} 
+	end
+
+	def player_won?
+		WIN_CONDITIONS.each do |val|
+			if  @board.matrix[val[0]] == @board.matrix[val[1]] && 
+			 	@board.matrix[val[1]] == @board.matrix[val[2]] &&
+			 	@board.matrix[val[0]] != nil && @board.matrix[val[1]] != nil && @board.matrix[val[2]] != nil
+			 	return true
+			end
+		end
+		return false
+	end
+
+	def change_names
+			puts "player1 name:"
+			@player1 = add_player('X')
+			puts "player2 name:"
+			@player2 = add_player('O')
+	end
```

Comment:
The name of this method is misleading. You create two player objects and assign them to a pair of instance variables. Instead, why not create a method that receives the player symbol, instantiates a player and then adds it to the game?